mitodl/mitxonline
# Adds a management command to create discount code(s) via command line

#### Pre-Flight checklist


#### What are the relevant tickets?

#1139 

#### What's this PR do?

Adds a command to create discount/enrollment codes via the command line. Optionally, it can create batches of codes.

#### How should this be manually tested?

Run the management command `generate_discount_code --help` to see the options, then test with various permutations:

* The defaults are to create a `percent-off` code with unlimited redemptions for the specified amount with the code specified. (You can specify multiple codes on the command line to generate several codes.) 
* You can add an expiration date by specifying `--expires`, or make the code a one-time redemption code by using `--one-time`
* The discount type can be specified with `--discount-type`
* Use `--prefix` and `--count` to generate a batch of codes. In this mode, the code specified on the command line will be ignored, and it will instead use your specified prefix and a UUID as the code. 

The command will write its output to `generated-codes.csv`. This file will be overwritten on each run.
